### PR TITLE
SRCH-184 add button to let super admins reindex a searchgov domain

### PR DESCRIPTION
**Pull request by MothOnMars**
This PR:
- adds a "Reindex" button to the super admin searchgov_domains table, enabling a super admin to trigger reindexing on a domain that has made major site changes
- add composite indices to searchgov_urls table to support common searches & scopes
- disables the Rubocop that checks for methods that skip ActiveRecord validations

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -2520,7 +2520,7 @@ Rails/SkipsModelValidations:
                  Use methods that skips model validations with caution.
                  See reference for more information.
   Reference: 'https://guides.rubyonrails.org/active_record_validations.html#skipping-validations'
-  Enabled: true
+  Enabled: false
```
Comment (MothOnMars) [inline .rubocop.yml:2523]: re-enable and add comment re. approving failures at the time of code review

File: spec/controllers/admin/searchgov_domains_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+
+describe Admin::SearchgovDomainsController do
```
Comment (MothOnMars) [inline spec/controllers/admin/searchgov_domains_controller_spec.rb:3]: convert to system spec

File: spec/controllers/admin/searchgov_domains_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  let(:searchgov_domain) { searchgov_domains(:basic_domain) }
+  let(:params) do
+    { id: searchgov_domain.id }
+  end
+
+  include_context 'super admin logged in' do
```
Comment (MothOnMars) [inline spec/controllers/admin/searchgov_domains_controller_spec.rb:9]: John sez - here there be tygers

File: app/controllers/omniauth_callbacks_controller.rb [hunk 4]
```diff
@@ -10,7 +10,8 @@ def login_dot_gov
     set_user_session
     redirect_to(admin_home_page_path)
   rescue LoginError => e
-    flash[:error] = "login internal error: #{e.message}"
+    flash[:error] = "login internal error: #{e.message}""login internal error: #{e.message}"
```
Comment (jmax-fearless) [inline app/controllers/omniauth_callbacks_controller.rb:13]: What? Looks like a typo to me.

File: spec/system/searchgov_domain_management.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    let(:current_user) { users(:affiliate_admin) }
+    before do
+      Rails.application.env_config["omniauth.auth"] = mock_user_auth('affiliate_admin@fixtures.org')
+    end
+
+    describe 'reindexing a domain' do
+      #subject(:reindex) { post :reindex, params: params }
+
+      it 'triggers a reindex of the domain' do
+        puts "visiting login"
```
Comment (jmax-fearless) [inline spec/system/searchgov_domain_management.rb:15]: Two things:
1) I _really_ don't like puts inside of specs. The extra noise during test runs is a problem.
2) Can we get rid of the steps that have nothing to do with the functionality under test? Login etc. aren't being tested here, and don't really belong in the test.
3) The test isn't actually testing anything; the expectation is commented out!

**Comment by MothOnMars** [comment]
Closing this up while I make additional changes to add the system spec.

**Comment by MothOnMars** [comment]
@jmax-fearless , this is ready for re-review. I've converted the searchgov domains controller spec to a system spec, & added a comment to the Rubocop cop for potentially dangerous update methods.

**Pull request merged**